app/src/components/protected.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext, useEffect } from 'react'
+import { AuthContext, AuthContextType } from 'STARTER/contexts/auth'
+import {
+  useNavigate
+} from 'react-router-dom'
+
+import CircularProgress from '@mui/material/CircularProgress'
+
+const Protected = ({ children }) => {
+  const { isAuthed, isAuthLoading } = useContext(AuthContext) as AuthContextType
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    if (!isAuthed && !isAuthLoading) navigate('/login')
+  })
```

Comment:
```suggestion
  }, [isAuthed, isAuthLoading])
```

We almost always want to include a dependencies list in `useEffect` hooks